test/jpypetest/test_jvmfinder.py

```diff
@@ -66,34 +64,113 @@ def test_get_from_bin(self, mock_real_path, mock_path_exists, mock_os_walk):
         mock_real_path.return_value = '/usr/lib/jvm/java-6-openjdk-amd64/bin/java'
 
-        finder = LinuxJVMFinder()
+        finder = jpype._jvmfinder.LinuxJVMFinder()
         p = finder._get_from_bin()
 
         self.assertEqual(
             p, os.path.join('/usr/lib/jvm/java-6-openjdk-amd64/jre/lib/amd64/server', 'libjvm.so'))
 
     @mock.patch('platform.mac_ver')
-    def test_javahome_binary_py27(self, mock_mac_ver):
+    def testDarwinBinary(self, mock_mac_ver):
         # this version has java_home binary
         mock_mac_ver.return_value = ('10.6.8', '', '')
 
         expected = '/System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home\n'
 
-        finder = DarwinJVMFinder()
+        finder = jpype._jvmfinder.DarwinJVMFinder()
 
         # fake check_output
         with mock.patch('subprocess.check_output') as mock_checkoutput:
             mock_checkoutput.return_value = expected
             p = finder._javahome_binary()
-
             self.assertEqual(p, expected.strip())
 
         # this version has no java_home binary
         mock_mac_ver.return_value = ('10.5', '', '')
         p = finder._javahome_binary()
-
         self.assertEqual(p, None)
 
+    @common.unittest.skip
+    def testPlatform(self):
+        with mock.patch('jpype._jvmfinder.sys') as mocksys, mock.patch('jpype._jvmfinder.WindowsJVMFinder') as finder:
+            mocksys.platform = 'win32'
+            jpype._jvmfinder.getDefaultJVMPath()
+            self.assertIn(finder().get_jvm_path, finder.mock_calls)
+
+        with mock.patch('jpype._jvmfinder.sys') as mocksys, mock.patch('jpype._jvmfinder.LinuxJVMFinder') as finder:
+            mocksys.platform = 'linux'
+            getDefaultJVMPath()
+            self.assertIn(finder().get_jvm_path, finder.mock_calls)
+
+        with mock.patch('jpype._jvmfinder.sys') as mocksys, mock.patch('jpype._jvmfinder.DarwinJVMFinder') as finder:
+            mocksys.platform = 'darwin'
+            getDefaultJVMPath()
+            self.assertIn(finder().get_jvm_path, finder.mock_calls)
+
+        with mock.patch('jpype._jvmfinder.sys') as mocksys, mock.patch('jpype._jvmfinder.CygwinJVMFinder') as finder:
+            mocksys.platform = 'cygwin'
+            getDefaultJVMPath()
+            self.assertIn(finder().get_jvm_path, finder.mock_calls)
+
+    def testLinuxGetFromBin(self):
+        finder = jpype._jvmfinder.LinuxJVMFinder()
+
+        def f(s):
+            return s
+        with mock.patch('os.path') as pathmock, \
+                mock.patch('jpype._jvmfinder.JVMFinder.find_libjvm') as mockfind:
+            pathmock.exists.return_value = True
+            pathmock.realpath = f
+            pathmock.abspath = f
+            pathmock.dirname = os.path.dirname
+            pathmock.join = os.path.join
+            finder._get_from_bin()
+            self.assertEqual(
+                pathmock.dirname.mock_calls[0].args, (finder._java,))
+
+    @common.unittest.skip
+    def testCheckArch(self):
```

Comment:
this tests needs the same (reverse) implementation like the actual impl. I would recommend against this pattern, since when you need to make a change you need to double implement it for the test as well. I'm curious if still there are good reasons to do so.

Reply:
Same as the previous.   To make sure we don't have a problem with a typo or accidental edit I recommend those tests.  Obviously they need to be updated with the code changes rather than just a fixed static test like the others.  A decorator should help make it clear and also make it so that it is only tested in the coverage phase.

Reply:
I don't see this point. It only adds more maintaining overhead + doesn't add new test cases as it only reproduces the implementation detail. Same as above, if you need to check for SyntaxErrors you need to run the code paths via an API call.